core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -290,18 +320,20 @@ private void setInstrumentationKey(ApplicationInsightsXmlConfiguration userConfi
 
         ArrayList<TelemetryModule> modules = new ArrayList<TelemetryModule>();
 
-        final List<String> performanceModuleNames =
-                new AnnotationPackageScanner().scanForClassAnnotations(new Class[]{PerformanceModule.class}, performanceCountersSection);
+        List<String> performanceModuleNames = new ArrayList<>();
+        if (System.getProperty(PERFORMANCE_MODULES_SCANNING_ENABLED_PROPERTY, "false").equalsIgnoreCase("true")) {
```

Comment:
This does change behavior, but I think this is ok. I don't think there are customers using the annotations and I would like to deprecate them (we can do this now, if you like). If someone does want these turned on, they can use the property.

Reply:
I am fine with reducing as much as code as possible and focus on simple and elegant API.  I have no objections and since we would be releasing a feature version 2.1.0, it might be a good time.